single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/positionfromextension/PositionFromExtension.java

```diff
@@ -117,17 +117,26 @@ public LBSCluster organizeLegBusSets(VoltageLevelGraph graph, List<LegBusSet> le
     }
 
     private void gatherLayoutExtensionInformation(VoltageLevelGraph graph) {
-        graph.getNodes().stream().filter(node -> node.getDirection() != Direction.UNDEFINED && graph.getCell(node).isPresent()).forEach(node -> {
-            BusCell cell = (BusCell) graph.getCell(node).get();
-            cell.setDirection(node.getDirection());
-            node.getOrder().ifPresent(cell::setOrder);
-        });
+
         graph.getBusCellStream().forEach(bc -> {
+            List<Direction> listOfDirectionsInsideCell = bc.getNodes().stream().map(Node::getDirection).distinct().collect(Collectors.toList());
+            int numberOfDirectionsInsideCell = listOfDirectionsInsideCell.size();
+            if (numberOfDirectionsInsideCell == 1) {
+                Direction direction = listOfDirectionsInsideCell.get(0);
+                if (direction != Direction.UNDEFINED) {
```

Comment:
you can remove that test: `BusCell` direction is `UNDEFINED` by default, so `bc.setDirection(UNDEFINED)` is not very harmful!

Reply:
If I remove the condition, some tests fail. There might be an underlying bug in the setDirection function. I need to check that further!